src/ripple/protocol/STObject.h

```diff
@@ -538,13 +538,22 @@ class STObject
     }
 
 private:
-    void add (Serializer & s, bool withSigningFields) const;
+    enum WhichFields : bool
+    {
+        // These values are carefully chosen to do the right thing if passed
+        // to SField::shouldInclude (bool)
+        withoutSigningFields = false,
+        withAllFields = true
```

Comment:
I think the distinction between `with` and `omit` is great.  I'm leaning toward
```
omitSigningFields = false,
withAllFields = true
```
I prefer `withAllFields` over `withSigningFields`, because it says what were doing.  We leave nothing out.  But that's just my take.  Will that work for you @seelabs?

Reply:
I'm fine with that.